lib/eventFactory.js

```diff
@@ -73,6 +73,32 @@ function getJobsFromPR(config) {
     return jobsToStart;
 }
 
+/**
+ * Starts the build if the changed file is part of the sourcePaths, or if there is no sourcePaths
+ * @method startBuild
+ * @param  {Object}   config                configuration object
+ * @param  {Object}   config.buildConfig    Build Config to create the build with
+ * @param  {Array}    config.changedFiles   List of files that were changed
+ * @param  {Array}    config.sourcePaths    List of soure paths
+ * @return {Promise}
+ */
+function startBuild(config) {
+    /* eslint-disable global-require */
+    const BuildFactory = require('./buildFactory');
+    const buildFactory = BuildFactory.getInstance();
+    /* eslint-enable global-require */
+    const { buildConfig, changedFiles, sourcePaths } = config;
+    let hasChangeInSourcePaths = true;
+
+    // Only check if sourcePaths exists
+    if (sourcePaths) {
+        hasChangeInSourcePaths = changedFiles.some(file =>
+            sourcePaths.some(source => file.includes(source)));
```

Comment:
What if `sourcePaths = ['src/folder']` and `changedFiles = ['src/folder/file.txt']`? Will this logic still work?

Reply:
Yes. that is the test case. https://github.com/screwdriver-cd/models/pull/232/files#diff-88f33fafc6c2f1254b793e6e56a1917aR458